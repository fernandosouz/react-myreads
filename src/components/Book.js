import React, { Component } from 'react';
import * as BooksAPI from "../Utils/BooksAPI";
import BookDetails from './BoookDetails';
import 'react-toastify/dist/ReactToastify.css';



class Book extends Component{

    state  = {
        book: '',
        arrayOptions: [
            {name: "currentlyReading", desc:"Currently Reading"},
            {name: "wantToRead", desc: "Want to Read"},
            {name: "read", desc: "Read"},
            {name: "none", desc: "None"}
        ],
        openModal: false} ;

    notifySucesss = () => {
        const x = document.getElementById("snackbar");
        x.className = "show";
        this.props.updateList(this.state.book);
        setTimeout(() => {
            x.className = x.className.replace("show", "");
            }, 2000);
    };

    notifyErr = () => {
        const x = document.getElementById("snackbarerror");
        x.className = "show";
        setTimeout(() => {
            x.className = x.className.replace("show", "");
        }, 2000);
    }

    updateAPI = (event) => {
        const b = this.props.book;
        b.shelf = event.target.value;
        this.setState({book: b});
        BooksAPI.update(this.props.book, event.target.value).then((response) => {
            this.notifySucesss();
        }, (response) => {
            console.log(response);
            this.notifyErr();
        });
    }

    componentDidMount(){
        this.setState({
            book: this.props.book
        })
    }

    render(){
        const {title, authors} = this.props.book;
        const {arrayOptions} = this.state;

        return(
            <div>
                <div className="card" style={{width: 180}}>
                    <img className="card-img-top" src={this.props.book.imageLinks.thumbnail} alt=""></img>
                        <div className="card-body" style={{padding: 5}}>
                            <h5 className="card-title book-title">{title}</h5>
                            <p className="card-text book-authors">{authors}</p>
                        </div>
                    <div className="row" style={{marginLeft: 0}}>
                        <BookDetails style={{width: 30, paddingLeft:10}} details={this.props.book} />
                        <div className="form-group" style={{width: 110}}>
                            <select style={{fontSize:10, padding:2}} className="form-control" id="exampleSelect1" value={this.props.book.shelf} onChange={this.updateAPI}>
                                {arrayOptions.map((obj) => (
                                    <option style={{fontSize:14}}
                                        key={obj.name}
                                        value={obj.name}>{obj.desc}</option>
                                ))}
                            </select>
                        </div>
                    </div>
                </div>
                <div id="snackbar">Successfully!</div>
                <div id="snackbarerror">Ops, try again!</div>
               {/* <div className="book">
                    <div className="book-top">
                        <div className="book-cover" style={{
                            width: 128,
                            height: 188,
                            backgroundImage: `url(${ this.props.book.imageLinks.thumbnail})`}}></div>
                        <BookDetails details={this.props.book} />
                        <div className="book-shelf-changer">
                            <select className="form-control" id="exampleSelect1" value={this.props.book.shelf} onChange={this.updateAPI}>
                                {arrayOptions.map((obj) => (
                                    <option
                                            key={obj.name}
                                            value={obj.name}>{obj.desc}</option>
                                ))}
                            </select>
                        </div>
                    </div>
                    <div className="book-title">{title}</div>
                    <div className="book-authors">{authors}</div>
                    <div>
                        <ToastContainer />
                    </div>
                </div>*/}

            </div>
        )
    }
}

export default Book
