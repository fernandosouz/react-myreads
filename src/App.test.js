import React from 'react';
import { shallow } from 'enzyme'
import Book from "./components/Book";
import BookList from "./components/BooksList";
import renderer from 'react-test-renderer';


describe('[Component] Book', () => {

    /*it('shallow render Book component correctly', () => {
        expect(shallow(<Book book={
            {shelf: 'none',
            imageLinks: {
                    thumbnail: "http://books.google.com/books/content?id=4O8Gfb76eTgC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
                }
            }
        } />));
    });*/

    it('[Book] renders correctly', () => {
        const tree = renderer.create(
            <Book book={
                {shelf: 'none',
                    imageLinks: {
                        thumbnail: "http://books.google.com/books/content?id=4O8Gfb76eTgC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
                    }
                }
            } />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('[BooksList] renders correctly', () => {
        const tree = renderer.create(
            <BookList list={[
                {shelf: 'none',
                    imageLinks: {
                        thumbnail: "http://books.google.com/books/content?id=4O8Gfb76eTgC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
                    }
                },
                {shelf: 'none',
                    imageLinks: {
                        thumbnail: "http://books.google.com/books/content?id=4O8Gfb76eTgC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
                    }
                }]}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });


});