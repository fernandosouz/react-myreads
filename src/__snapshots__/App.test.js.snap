// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Book [Book] renders correctly 1`] = `
<div>
  <div
    className="card"
    style={
      Object {
        "width": 180,
      }
    }
  >
    <img
      alt=""
      className="card-img-top"
      src="http://books.google.com/books/content?id=4O8Gfb76eTgC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
    />
    <div
      className="card-body"
      style={
        Object {
          "padding": 5,
        }
      }
    >
      <h5
        className="card-title book-title"
      />
      <p
        className="card-text book-authors"
      />
    </div>
    <div
      className="row"
      style={
        Object {
          "paddingLeft": "18%",
          "width": "100%",
        }
      }
    >
      <div>
        <button
          className="btn btn-outline-primary"
          onClick={[Function]}
          style={
            Object {
              "fontSize": 12,
              "height": "69%",
            }
          }
          type="button"
        >
          More
        </button>
      </div>
      <div
        className="form-group"
        style={
          Object {
            "width": 90,
          }
        }
      >
        <select
          className="form-control"
          id="exampleSelect1"
          onChange={[Function]}
          style={
            Object {
              "fontSize": 10,
              "padding": 2,
            }
          }
          value="none"
        >
          <option
            style={
              Object {
                "fontSize": 14,
              }
            }
            value="currentlyReading"
          >
            Currently Reading
          </option>
          <option
            style={
              Object {
                "fontSize": 14,
              }
            }
            value="wantToRead"
          >
            Want to Read
          </option>
          <option
            style={
              Object {
                "fontSize": 14,
              }
            }
            value="read"
          >
            Read
          </option>
          <option
            style={
              Object {
                "fontSize": 14,
              }
            }
            value="none"
          >
            None
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    id="snackbar"
  >
    Successfully!
  </div>
  <div
    id="snackbarerror"
  >
    Ops, try again!
  </div>
</div>
`;

exports[`[Component] Book [BooksList] renders correctly 1`] = `
<div>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <li>
        <div>
          <div
            className="card"
            style={
              Object {
                "width": 180,
              }
            }
          >
            <img
              alt=""
              className="card-img-top"
              src="http://books.google.com/books/content?id=4O8Gfb76eTgC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
            />
            <div
              className="card-body"
              style={
                Object {
                  "padding": 5,
                }
              }
            >
              <h5
                className="card-title book-title"
              />
              <p
                className="card-text book-authors"
              />
            </div>
            <div
              className="row"
              style={
                Object {
                  "paddingLeft": "18%",
                  "width": "100%",
                }
              }
            >
              <div>
                <button
                  className="btn btn-outline-primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "fontSize": 12,
                      "height": "69%",
                    }
                  }
                  type="button"
                >
                  More
                </button>
              </div>
              <div
                className="form-group"
                style={
                  Object {
                    "width": 90,
                  }
                }
              >
                <select
                  className="form-control"
                  id="exampleSelect1"
                  onChange={[Function]}
                  style={
                    Object {
                      "fontSize": 10,
                      "padding": 2,
                    }
                  }
                  value="none"
                >
                  <option
                    style={
                      Object {
                        "fontSize": 14,
                      }
                    }
                    value="currentlyReading"
                  >
                    Currently Reading
                  </option>
                  <option
                    style={
                      Object {
                        "fontSize": 14,
                      }
                    }
                    value="wantToRead"
                  >
                    Want to Read
                  </option>
                  <option
                    style={
                      Object {
                        "fontSize": 14,
                      }
                    }
                    value="read"
                  >
                    Read
                  </option>
                  <option
                    style={
                      Object {
                        "fontSize": 14,
                      }
                    }
                    value="none"
                  >
                    None
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            id="snackbar"
          >
            Successfully!
          </div>
          <div
            id="snackbarerror"
          >
            Ops, try again!
          </div>
        </div>
      </li>
      <li>
        <div>
          <div
            className="card"
            style={
              Object {
                "width": 180,
              }
            }
          >
            <img
              alt=""
              className="card-img-top"
              src="http://books.google.com/books/content?id=4O8Gfb76eTgC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
            />
            <div
              className="card-body"
              style={
                Object {
                  "padding": 5,
                }
              }
            >
              <h5
                className="card-title book-title"
              />
              <p
                className="card-text book-authors"
              />
            </div>
            <div
              className="row"
              style={
                Object {
                  "paddingLeft": "18%",
                  "width": "100%",
                }
              }
            >
              <div>
                <button
                  className="btn btn-outline-primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "fontSize": 12,
                      "height": "69%",
                    }
                  }
                  type="button"
                >
                  More
                </button>
              </div>
              <div
                className="form-group"
                style={
                  Object {
                    "width": 90,
                  }
                }
              >
                <select
                  className="form-control"
                  id="exampleSelect1"
                  onChange={[Function]}
                  style={
                    Object {
                      "fontSize": 10,
                      "padding": 2,
                    }
                  }
                  value="none"
                >
                  <option
                    style={
                      Object {
                        "fontSize": 14,
                      }
                    }
                    value="currentlyReading"
                  >
                    Currently Reading
                  </option>
                  <option
                    style={
                      Object {
                        "fontSize": 14,
                      }
                    }
                    value="wantToRead"
                  >
                    Want to Read
                  </option>
                  <option
                    style={
                      Object {
                        "fontSize": 14,
                      }
                    }
                    value="read"
                  >
                    Read
                  </option>
                  <option
                    style={
                      Object {
                        "fontSize": 14,
                      }
                    }
                    value="none"
                  >
                    None
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            id="snackbar"
          >
            Successfully!
          </div>
          <div
            id="snackbarerror"
          >
            Ops, try again!
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;
